server/handlers/clienthandler.go

```diff
@@ -43,18 +45,23 @@ func GetClient(w http.ResponseWriter, r *http.Request) {
 }
 
 func CreateClient(w http.ResponseWriter, r *http.Request) {
-	var client model.Client
-	if err := json.NewDecoder(r.Body).Decode(&client); err != nil {
+	var clientReq request.ClientRequest
+	if err := json.NewDecoder(r.Body).Decode(&clientReq); err != nil {
 		log.Print("client decode malfunction")
 		w.WriteHeader(400)
 		fmt.Fprintln(w, "an error occurred creating client")
 		return
 	}
-	clientId, err := repository.CreateClient(&client) 
-	if err != nil {
+	client, err := validation.ClientValidation(&clientReq); if err != nil {
+		log.Println("client not created")
+		w.WriteHeader(400)
+		fmt.Fprintf(w, "an error occurred creating client: %v", err)
```

Comment:
validation errors should return 422 and a message stating that it's specifically a validation error